azurerm/resource_arm_app_service.go

```diff
@@ -0,0 +1,642 @@
+			SiteConfig:   &siteConfig,
+		},
+	}
+
+	if v, ok := d.GetOk("client_affinity_enabled"); ok {
+		enabled := v.(bool)
+		siteEnvelope.SiteProperties.ClientAffinityEnabled = utils.Bool(enabled)
+	}
+
+	// NOTE: these seem like sensible defaults, in lieu of any better documentation.
+	skipDNSRegistration := false
+	forceDNSRegistration := false
+	skipCustomDomainVerification := true
+	ttlInSeconds := "60"
+	_, createErr := client.CreateOrUpdate(resGroup, name, siteEnvelope, &skipDNSRegistration, &skipCustomDomainVerification, &forceDNSRegistration, ttlInSeconds, make(chan struct{}))
+	err := <-createErr
+	if err != nil {
+		return err
+	}
+
+	settings := web.StringDictionary{
+		Properties: appSettings,
+	}
+	_, err = client.UpdateApplicationSettings(resGroup, name, settings)
+	if err != nil {
+		return fmt.Errorf("Error updating Application Settings for App Service %q: %+v", name, err)
+	}
+
+	connectionStrings := expandAppServiceConnectionStrings(d)
+	properties := web.ConnectionStringDictionary{
+		Properties: connectionStrings,
+	}
+
+	_, err = client.UpdateConnectionStrings(resGroup, name, properties)
+	if err != nil {
+		return fmt.Errorf("Error updating Connection Strings for App Service %q: %+v", name, err)
+	}
+
+	read, err := client.Get(resGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read App Service %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmAppServiceUpdate(d, meta)
+}
+
+func resourceArmAppServiceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).appServicesClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["sites"]
+
+	if d.HasChange("site_config") {
+		// update the main configuration
+		siteConfig := expandAppServiceSiteConfig(d)
+		siteConfigResource := web.SiteConfigResource{
+			SiteConfig: &siteConfig,
+		}
+		_, err := client.CreateOrUpdateConfiguration(resGroup, name, siteConfigResource)
+		if err != nil {
+			return fmt.Errorf("Error updating Configuration for App Service %q: %+v", name, err)
+		}
+	}
+
+	if d.HasChange("app_settings") {
+		// update the AppSettings
+		appSettings := expandAppServiceAppSettings(d)
+		settings := web.StringDictionary{
+			Properties: appSettings,
+		}
+
+		_, err := client.UpdateApplicationSettings(resGroup, name, settings)
+		if err != nil {
+			return fmt.Errorf("Error updating Application Settings for App Service %q: %+v", name, err)
+		}
+	}
+
+	if d.HasChange("connection_string") {
+		// update the ConnectionStrings
+		connectionStrings := expandAppServiceConnectionStrings(d)
+		properties := web.ConnectionStringDictionary{
+			Properties: connectionStrings,
+		}
+
+		_, err := client.UpdateConnectionStrings(resGroup, name, properties)
+		if err != nil {
+			return fmt.Errorf("Error updating Connection Strings for App Service %q: %+v", name, err)
+		}
+	}
+
```

Comment:
do any of these need to check to make sure it's Not a new resource?

Reply:
nope, they're all applicable here